nsv13/code/modules/munitions/ship_weapons/volatility_component.dm

```diff
@@ -59,3 +61,34 @@ Add this component to an atom to mark it as volatile, if it takes fire damage, i
 	UnregisterSignal(parent, COMSIG_ATOM_FIRE_ACT)
 	UnregisterSignal(parent, COMSIG_PARENT_EXAMINE)
 	return ..()
+
+/datum/component/volatile/emptorp
+	desc = "<span class='warning'>It's highly volatile and risks misfires if subjected to heat!</span>"
+
+///Overrides parent proc.
+/datum/component/volatile/emptorp/explode()
+	if(!parent)
+		message_admins("Volatility component tried to explode with no attached parent. Contact a coder")
+		return FALSE
+	log_game("Volatile substance caused an electromagnetic reaction at [get_area(parent)]")
+	var/base_ex_power = explosion_scale * volatility
+	empulse(get_turf(parent), base_ex_power, base_ex_power * 2)
+	explosion(parent, 0, 1, round(base_ex_power / 2), smoke = TRUE)
+	if(!QDELETED(parent))
+		qdel(parent)
+
+/datum/component/volatile/helltorp //Hoo boy
+	desc = "<span class='warning'>It's extremely volatile and prone to runaway reactions if subjected to heat!</span>"
+
+///Overrides parent proc
+/datum/component/volatile/helltorp/explode()
+	if(!parent)
+		message_admins("Volatility component tried to explode with no attached parent. Contact a coder")
+		return FALSE
+	log_game("Volatile substance caused a runaway hellfire reaction at [get_area(parent)]")
+	var/base_ex_power = explosion_scale * volatility
+	explosion(parent, 0, 0, base_ex_power, base_ex_power * 2, base_ex_power)
+	var/turf/detonation_turf = get_turf(parent)
+	detonation_turf.atmos_spawn_air("o2=20;plasma=110;TEMP=700")
+	if(!QDELETED(parent))
+		qdel(parent)
```

Comment:
smh this is not how you're supposed to use components. Components should be generic, special implementation should always be done on the object

Reply:
I mean this is kinda what oop exists for no?
Is there any actual practical disadvantage, considering subtyping these prevents having to copypaste behaviors for projectiles with simillar volatility reactions but different inheritance trees?

Reply:
Components are ECS, not OOP